README.md

```diff
@@ -45,24 +45,78 @@ Returns source map converter from given base64 encoded json string.
 
 Returns source map converter from given base64 or uri encoded json string prefixed with `//# sourceMappingURL=...`.
 
-### fromMapFileComment(comment, mapFileDir)
+### fromMapFileComment(comment, readMap)
 
 Returns source map converter from given `filename` by parsing `//# sourceMappingURL=filename`.
 
-`filename` must point to a file that is found inside the `mapFileDir`. Most tools store this file right next to the
-generated file, i.e. the one containing the source map.
+`readMap` must be a function which receives the source map filename and returns either a String or Buffer of the source map (if read synchronously), or a `Promise` containing a String or Buffer of the source map (if read asynchronously).
+
+If `readMap` doesn't return a `Promise`, `fromMapFileComment` will return a source map converter synchronously.
+
+If `readMap` returns a `Promise`, `fromMapFileComment` will also return `Promise`. The `Promise` will be either resolved with the source map converter or rejected with an error.
+
+#### Examples
+
+**Synchronous read in Node.js:**
+
+```js
+var convert = require('convert-source-map');
+var fs = require('fs');
+
+function readMap(filename) {
+  return fs.readFileSync(filename, 'utf8');
+}
+
+var json = convert
+  .fromMapFileComment('//# sourceMappingURL=map-file-comment.css.map', readMap)
+  .toJSON();
+console.log(json);
+```
+
+
+**Asynchronous read in Node.js:**
+
+```js
+var convert = require('convert-source-map');
+var fs = require('fs/promises'); // Notice the `fs/promises` import
```

Comment:
This is only supported in latest Node.js.

Might be better to change this as follows:
```suggestion
const { promises: fs } = require('fs'); // Notice the `promises` import
```